acvm/src/lib.rs

```diff
@@ -160,5 +170,18 @@ pub trait ProofSystemCompiler {
         public_inputs: WitnessMap,
         circuit: &Circuit,
         verification_key: &[u8],
+        is_recursive: bool,
     ) -> Result<bool, Self::Error>;
+
+    fn proof_as_fields(
```

Comment:
Why is this and the below function needed?

Reply:
We need a proof and verification key to be witnesses of the circuit in order to recursively verify them. Most proofs and verification key's are not formatted in a proof system's native field elements. I was debating to put this in `acvm-backend-barretenberg` but it felt universal enough for being able to perform recursion with Noir in general. 

@zac-williamson do you think these methods are too specific for the acvm and should possibly just be moved into our backend? For example, if a halo2 backend were to implement recursion do you think these methods would still be necessary?

Reply:
How/Where would these two methods be called?

Reply:
These would be called by someone looking to pass a proof and/or verification key into their Noir circuit. The way our recursive verification opcode is setup we require a verification key and a proof and these must be represented by witness values. 

Any backend is going to have its own recursion format for a proof and/or vkeys. Being able to export the proof and verification key in the appropriate format will be necessary for any backend looking to integrate recursion. I could move this functionality to acvm-backend-bberg but then I will still want to expose functionality for a backend to be able to generate the appropriate recursive format for a proof and vkey. 

Having these methods be a part of the ACVM allows for us to more easily integrate it into `nargo`. Otherwise this will be a purely backend specific functionality, when it really is not. 

Let's say I have circuit A. I compile and generate a proof for this circuit.

I want to verify circuit A in circuit B. I need to pass in the proof and vkey in the appropriate recursion format. These methods will be used to generate the proof and vkey as fields (perhaps I should rename the functions `***_in_recursion_format`?). The outputted proof and vkey as fields will be input to circuit B and used as parameters for the verify proof builtin.

Reply:
I added a comment on this function. It might not be descriptive enough so let me know what you think